core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -316,8 +341,10 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
    * Forcefully free the buffer's mmap.
    */
   protected[log] def forceUnmap() {
-    try MappedByteBuffers.unmap(file.getAbsolutePath, mmap)
-    finally mmap = null // Accessing unmapped mmap crashes JVM by SEGV so we null it out to be safe
+    if (mmap != null) {
```

Comment:
doing the NPE check fixes it. Unsure why `mmap` in the first place is null, probably this method might have been called twice? Anyway, I don't think it hurts to check for NPE in that case

```
[2018-09-03 00:16:04,902] INFO Updated PartitionLeaderEpoch. New: {epoch:0, offset:0}, Current: {epoch:-1, offset:-1} for Partition: test4-0. Cache now contains 0 entries. (kafka.server.epoch.LeaderEpochFileCache)
[2018-09-03 00:16:53,361] INFO Deleted log C:\tmp\kafka-logs\test3-0.d2d44094e2a74667b84ae6d843835a57-delete\00000000000000000000.log. (kafka.log.LogSegment)
[2018-09-03 00:16:53,361] INFO Deleted offset index C:\tmp\kafka-logs\test3-0.d2d44094e2a74667b84ae6d843835a57-delete\00000000000000000000.index. (kafka.log.LogSegment)
[2018-09-03 00:16:53,361] INFO Deleted time index C:\tmp\kafka-logs\test3-0.d2d44094e2a74667b84ae6d843835a57-delete\00000000000000000000.timeindex. (kafka.log.LogSegment)
[2018-09-03 00:16:53,375] INFO Deleted log for partition test3-0 in C:\tmp\kafka-logs\test3-0.d2d44094e2a74667b84ae6d843835a57-delete. (kafka.log.LogManager)
```

Reply:
We shouldn't do this, we need to understand why we are calling it twice as that might be a bug.

Reply:
to me this is because of https://github.com/apache/kafka/pull/5603/files#diff-ecef8994dd98759052b998e43fffc5b8R277